tests/extensions/test_version.py

```diff
@@ -104,6 +105,39 @@ def test_all_links(self):
         self.item.ext.version.apply(self.version, deprecated, latest, predecessor, successor)
         self.item.validate()
 
+    def test_full_copy(self):
```

Comment:
@schwehr here are the two tests that I used to make sure the object links were operating correctly in a full_copy scenario.

Notice that if you don't return the correct reltypes from the `_object_link` method, these tests fail.

This is because, when making a full copy of the catalog in memory, PySTAC doesn't know that those links point to STAC objects that should be considered in the copy. In [this bit of code](https://github.com/stac-utils/pystac/blob/develop/pystac/stac_object.py#L413-L427), the logic to determine whether or not a link has to be copied (and it's target either copied or pulled from the cache of already-copied objects) is determined by the _object_links method, which [as seen here](https://github.com/stac-utils/pystac/blob/develop/pystac/item.py#L265-L266) depends on object link the Catalog or Item declares (core reltypes like child, item or collection), as well as any reltypes that are declared to be STAC objects by implemented extensions.

Let me know if this helps explain the _object_links method and why this is a good way to test that, or feel free to make changes or give ideas on way to improve!

Reply:
Arg... I'm a noob and forced commit over your change.  I'll recommit it from this :(

Reply:
No worries! I find `git cherry-pick` to be the most useful in those types of situations.

Reply:
Thanks!  I spend most of my time in [piper](https://cacm.acm.org/magazines/2016/7/204032-why-google-stores-billions-of-lines-of-code-in-a-single-repository/fulltext) where cherry-picks are usually just for release branches.  Appreciate any help leveling up my git skills.